crates/core/c8y_api/src/smartrest/topic.rs

```diff
@@ -101,6 +102,21 @@ impl From<&EntityMetadata> for C8yTopic {
     }
 }
 
+impl From<&C8yAlarm> for C8yTopic {
+    fn from(value: &C8yAlarm) -> Self {
+        match value {
+            C8yAlarm::Create(alarm) => match alarm.source.as_ref() {
+                None => Self::SmartRestResponse,
+                Some(info) => Self::ChildSmartRestResponse(info.id.clone()),
+            },
+            C8yAlarm::Clear(alarm) => match alarm.source.as_ref() {
+                None => Self::SmartRestResponse,
+                Some(info) => Self::ChildSmartRestResponse(info.id.clone()),
+            },
+        }
+    }
+}
```

Comment:
This will have to be updated, once merged this PR: https://github.com/thin-edge/thin-edge.io/pull/2266#discussion_r1330224805

Reply:
Also, alarm won't use this after #2242 is done.